### DISCUSSION TITLE
Add MaybeNullIfNullAttribute to round out nullable annotations

### DISCUSSION THREAD

**Discussion by mikernet** [dotnet/csharplang#6981] [Language Ideas]
One notably absent annotation missing is `[MaybeNullIfNull]` as the inverse to `[NotNullIfNotNull]`. One use-case would be to allow mimicking NVT lifted operator behavior with a reference type:

```cs
[return: MaybeNullIfNull(nameof(x))]
[return: MaybeNullIfNull(nameof(y))]
public static RefValue operator +(RefValue? x, RefValue? y)
{
    if (x is null || y is null)
        return null;

    // ...
}
```

**Comment by RikkiGibson** [comment]
I think the scenario might be addressed by dotnet/roslyn#43906
